src/processor/common.ts

```diff
@@ -31,3 +36,26 @@ export function switchQuantityRules(rules: ExportableSdRule[]): void {
     }
   });
 }
+
+export function makeNameSushiSafe(
+  entity: ExportableProfile | ExportableExtension | ExportableValueSet | ExportableCodeSystem
+) {
+  if (/\s/.test(entity.name)) {
+    let entityType: string;
+    if (entity instanceof ExportableProfile || entity instanceof ExportableExtension) {
+      entityType = 'StructureDefinition';
+    } else if (entity instanceof ExportableValueSet) {
+      entityType = 'ValueSet';
+    } else {
+      entityType = 'CodeSystem';
+    }
+    logger.warn(
+      `${entityType} with id ${entity.id} has name with whitespace. Converting whitespace to underscores.`
+    );
+    const nameRule = new ExportableCaretValueRule('');
+    nameRule.caretPath = 'name';
+    nameRule.value = entity.name;
+    entity.name = entity.name.replace(/\s/g, '_');
+    entity.rules.unshift(nameRule);
+  }
+}
```

Comment:
Ha.  Yeah, you're right.  This is a lot simpler (and likely more elegant) than all those TypeScript shenanigans we were thinking of yesterday.  Good thinking!